cip/1.accepted/CIP2017-01-18-configurable-pattern-matching-semantics.adoc

```diff
@@ -93,26 +100,68 @@ Note that path patterns are assumed from the *CIP2017-02-06 Path Pattern Queries
 .Overview of proposed grammar changes:
 [source,ebnf]
 ----
-Pattern = PatternPart, { ',', PatternPart } ;
-PatternPart = [ PatternVariableSpec ], NodePattern, { (EdgePattern | PathPattern), NodePattern } ;
+!! Clause-level argument
+Pattern = PatternGroup, { PatternGroupWithModifiers }
+        | { PatternGroupWithModifiers }-
+        ;
 
-PatternVariableSpec = [ PatternMatchMode ], [ PatternVariableClass ], [ Variable, '=' ] ;
+PatternGroup = PatternPart, { ',', PatternPart } ;
 
-PatternMatchMode     = 'ALL'
-                     | 'ALL SHORTEST'
-                     | 'SHORTEST'
-                     ;
-PatternVariableClass = ([ PVClassModifier ], 'WALK', ['S'])
-                     | ([ PVClassModifier ], 'TRAIL', ['S'])
-                     | ([ PVClassModifier ], 'PATH', ['S'])
-                     ;
-PVClassModifier      = 'OPEN'
-                     | 'CLOSED'
+!! Expected to form a connected component
+PatternPart = NamedPatternPart
+            | AnonymousPatternPart
+            ;
+
+NamedPatternPart = PatternVariableSpec, AnonymousPatternPart ;
+
+AnonymousPatternPart = {
+                         NodePattern,
+                         { (EdgePattern | PathPattern), NodePattern }
+                       }-
                      ;
-----
 
+PathVariableSpec = [ [ PathClassModifier ], PathClass ], [ Variable, '=' ] ;
+
+PathClassModifier = ('OPEN' | 'CLOSED'), 'DIRECTED'
+                  | 'DIRECTED'
+                  ;
 
-As we can see above, patterns in Cypher consist of a comma-separated list of _pattern parts_, where a pattern part is exemplified by `p = (e:Employee)-[:REPORTS_TO*1..3]->(m:Manager)`.
+PathClass = 'WALK'
+          | 'TRAIL'
+          | [ 'SIMPLE' ], 'PATH'
```

Comment:
Does `TRAIL`, `PATH` and `SIMPLE PATH` really encode three different classes? If not I wonder why synonyms are allowed. (a `WALK` is obviously different from those three)